Shopify/dawn
# Logical properties - support for RTL

**PR Summary:** 

This PR changes lots of our CSS so that it uses logical properties instead. 

**Why are these changes introduced?**

This is introduced to make our theme work both in a Left To Right and Right To Left situation. This way merchants using languages that reads from right to left can use our theme. At the moment if a merchant needs to change the reading direction they will most likely contact support to be told that it's not a change we support. 

We used to support that kinda change but on Dawn we're not leveraging any framework or existing library. So when changing the website's reading direction the changes necessary are quite big. 

Logical properties are meant to solve this issue which is why it'd be great to use them going forward. 

**What approach did you take?**

Here is a recap of what they are: 
|  New property | Old one   |
|---|---|
| block-size | height |
| inline-size | width |
| border-block-start | `border-top` |
| border-block-end | `border-bottom` |
| border-inline-start | `border-left` |
| border-inline-end | `border-right` |
| margin-block-start | `margin-top` |
| margin-block-end | `margin-bottom` |
| margin-inline-start | `margin-left` |
| margin-inline-end | `margin-right` |
| padding-block-start | `padding-top` |
| padding-block-end | `padding-bottom` |
| padding-inline-start | `padding-left` |
| padding-inline-end | `padding-right` |
| inset-block-start | `top` |
| inset-block-end | `bottom` |
| inset-inline-start | `left` |
| inset-inline-end | `right` |
| inline-start | value for `float/clear` |
| inline-end | value for `float/clear` |
| start | value for `text-align: left` |
| end | Value for `text-align: right` |


**Here I focus on using only the `margin-inline`, `padding-inline`, `text-align`, `border-inline` logical properties as they are the one that are supported the most. Going all the way back to iOS 12.2 and being supported at 94% by browsers (based on [caniuse.com](caniuse.com))**

I have not changed some original properties in some occasions: 
- both `left` and `right` properties have the same value applied. So the direction shouldn't matter. 
- in some scenarios with have some absolute positioning where we center things using `transform` so I don't need to tweak the values either. 

**Other considerations**

**Testing steps/scenarios**
 
<details><summary>To test, I added a general setting to enable/disable RTL.</summary>

![](https://screenshot.click/12-25-zr5wv-bs7eu.png) 
</details>

- [ ] Test all sections in both LTR and RTL
- [ ] Test different section settings to make sure it doesn't break any previous behaviour
- [ ] Things that are using an absolute positioning and a `transform: translate(50%)` to center them shouldn't be using logical properties. I might have applied it to some. I caught a few that I fixed already.
- [ ] Test with all global settings coming up (shadows, borders, etc)

Here are a couple sites that use RTL and confirm certain patterns (close icon showing at the top left vs the top right in LTR): https://hotcrown.com/ & https://shlomitofir.com/

**Demo links**

- [Store](https://os2-demo.myshopify.com/?preview_theme_id=128347439126)
- [Editor](https://os2-demo.myshopify.com/admin/themes/128347439126/editor)

**Checklist**
